python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -718,4 +727,18 @@ def measure(self, measRecord, exposure, refRecord, refWcs):
         # Docstring inherited.
         wcs = exposure.getWcs()
         center = wcs.skyToPixel(refWcs.pixelToSky(refRecord.getCentroid()))
+        if self.config.doOptimalPhotometry:
+            # The adaptive shape is set to post-seeing aperture.
+            # Convolve it with the PSF shape to obtain pre-seeing aperture.
+            # Refer to pg. 30-31 of Kuijken et al. (2015) for this heuristic.
```

Comment:
They are not identical heuristic, but have the same idea. The code follows these sentences from the paper

> While for isolated sources any aperture matrix can be used to obtain unbiased estimates of the GAaP flux, the optimal SNR is obtained when the shape and orientation of the post-seeing aperture defined by W− p^2**1** is matched to the source.

So I assign the adaptive shape from the reference catalog, assign it to W_\text{int} - W_\text{PSF} and add the measured PSF shape to obtain W_\text{int}. In KiDS, not all the 9 bands were available to begin with (i.e., no `refCat`), so some proxy for PSF shape was added to the adaptive shape. Since our code can be used by any other observatory, I did not want to use any hard coded seeing like 0.7" and using the available PSF shape from `refCat` makes the most sense. As Konrad and I both find that the increase in SNR is only marginal, any variant of this heuristic works. We can always improve the heuristic in a later (unspecified) ticket, while keeping the framework if we find the performance is not satisfactory on the data.
 
 Finally, the upper limit of having 2" is also not required since the measurement happens on deblended images and we do not have to worry about picking light from the neighbors.